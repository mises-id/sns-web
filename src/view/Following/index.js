/*
 * @Author: lmk
 * @Date: 2021-07-15 13:41:35
 * @LastEditTime: 2021-08-08 00:12:01
 * @LastEditors: lmk
 * @Description: Following and Followers page
 */
import Cell from '@/components/Cell';
import React, {  useState } from 'react';
import { useTranslation } from 'react-i18next';
import followed from '@/images/followed.png' //mutual following
// import follow from '@/images/follow.png' //follow
// import following from '@/images/isfollow.png' //following
import './index.scss';
import { useLocation } from 'react-router-dom';
import { friendShip } from '@/api/fans';
import { useSelector } from 'react-redux';
import PullList from '@/components/PullList';
import Navbar from '@/components/NavBar';
const Following = ({history})=>{
  const {t} = useTranslation();
  const location = useLocation();
  const {state={}} = location || {}
  const type = state.pageType || 'following'
  const pageTitle = `${type}PageTitle`
  const [dataSource, setDataSource] = useState([]);
  const user = useSelector(state => state.user)
  const renderView =(val={},index)=>{
    return <Cell iconSize={35} className="m-bg-fff m-padding-lr15 m-padding-tb12" showArrow={false} label={val.username} key={index}
    rightChild={<img className="followedIcon" src={followed} alt="followIcon"  icon={val.avatar.medium}/>}></Cell>
  }
  //getData
  const fetchData = async () => {
    try {
      const res = await friendShip({
        uid:user.loginForm.uid,
        relate:type,
        page:1
      })
      const userList = res.data.map(val=>val.user)
      setDataSource([...dataSource,...userList]);
      return Promise.resolve(res)
    } catch (error) {
      return Promise.reject(222)
    }
  };
  return <div>
    <Navbar title={t(pageTitle)}
    />
    <PullList renderView={renderView} data={dataSource} load={fetchData}></PullList>
  </div>
}
export default Following