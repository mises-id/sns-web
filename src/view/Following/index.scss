.followIcon{
  width: 50px;
  height: 25px;
}