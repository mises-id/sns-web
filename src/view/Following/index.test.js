/*
 * @Author: lmk
 * @Date: 2021-07-15 16:07:05
 * @LastEditTime: 2022-03-08 09:38:01
 * @LastEditors: lmk
 * @Description: Following test
 */
import React from 'react';
import Following from './';
import {shallow } from 'enzyme';
describe('Following', () => {
  it('should render correctly', () => {
    const component = shallow(<Following />)
    expect(component).toMatchSnapshot();
  });
});