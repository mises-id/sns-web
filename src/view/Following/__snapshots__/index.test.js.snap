// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Following should render correctly 1`] = `
<div>
  <Navbar
    title="followingPageTitle"
  />
  <ForwardRef(PullList)
    data={Array []}
    load={[Function]}
    renderView={[Function]}
  />
</div>
`;
